### PR TITLE
Linmeta

### PR DESCRIPTION
**Pull request by ghost**
need an opinion on this method of fixing up the per-metric unit types

### CODE REVIEW

File: collectors/processes_linux.go [hunk 1]
```diff
@@ -21,18 +21,17 @@ func WatchProcesses(procs []*WatchedProc) error {
 }
 
 func linuxProcMonitor(w *WatchedProc, md *opentsdb.MultiDataPoint) error {
-	var err error
```
Comment (madelynnblue) [inline collectors/processes_linux.go]: What is all of this for?

File: collectors/disk_linux.go [hunk 2]
```diff
@@ -18,18 +18,24 @@ func init() {
...
-	"ios_in_progress",     // Number of actual I/O requests currently in flight.
-	"msec_total",          // Amount of time during which ios_in_progress >= 1.
-	"msec_weighted_total", // Measure of recent I/O completion time and backlog.
+type Metatuple struct {
+	key      string
+	ratetype metadata.RateType
+	unit     metadata.Unit
+}
+
+var FIELDS_DISK = []Metatuple{
```
Comment (madelynnblue) [inline collectors/disk_linux.go]: Change the var name to something like diskLinukFields.

File: collectors/disk_linux.go [hunk 3]
```diff
@@ -18,18 +18,24 @@ func init() {
...
-	"read_sectors",        // Total number of sectors read successfully.
-	"msec_read",           // Total number of ms spent by all reads.
-	"write_requests",      // Total number of writes completed successfully.
-	"write_merged",        // Adjacent write requests merged in a single req.
-	"write_sectors",       // Total number of sectors written successfully.
-	"msec_write",          // Total number of ms spent by all writes.
-	"ios_in_progress",     // Number of actual I/O requests currently in flight.
-	"msec_total",          // Amount of time during which ios_in_progress >= 1.
-	"msec_weighted_total", // Measure of recent I/O completion time and backlog.
+type Metatuple struct {
```
Comment (madelynnblue) [inline collectors/disk_linux.go]: I think we can make this type anonymous.

File: collectors/disk_linux.go (hunk 2 above)
Comment (madelynnblue) [inline collectors/disk_linux.go]:
```
var F = []struct {
    key  string
    rate metadata.RateType
    unit metadata.Unit
}{
    {"read_sectiors", ...},
...
}
```

File: collectors/processes_linux.go (hunk 1 above)
Comment (ghost) [inline collectors/processes_linux.go]: fix compile errors, I kind of guessed on a few of the error return logic

```
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:30: not enough arguments to return
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:35: not enough arguments to return
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:56: undefined: metadata.Fault
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:57: undefined: metadata.Fault
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:62: undefined: metadata.Syscall
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:63: undefined: metadata.Syscall
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:67: type error is not an expression
src/github.com/bosun-monitor/scollector/collectors/processes_linux.go:126: undefined: lp
```

Comment (madelynnblue) [inline collectors/processes_linux.go]: I probably need to fix some things first then. Hold up.

**Pull request merged**